### PR TITLE
Pathing updates

### PR DESCRIPTION
**Pull request by thorrester**
### Short Summary
Updates the cli to be compatible with current `opsml`
- Updates metric pathing
- Updates deps for sec vulns
- updates pathing creation for nested directories

### CODE REVIEW

File: src/api/metrics.rs [hunk 1]
```diff
@@ -194,43 +87,21 @@ impl MetricGetter {
...
-    // set up repair request
-    let compare_mertic = MetricGetter {};
-
-    compare_mertic
-        .compare_model_metrics(metric_name, lower_is_better, challenger_uid, champion_uid)
-        .await
+    metric_getter.get_model_metrics(uid).await
 }
 
 #[cfg(test)]
```
Comment (russellkemmit) [inline src/api/metrics.rs:95]: I am not a fan of housing tests within the same files as the mod itself. Not sure why Rust folks like this lol?

**Review by russellkemmit — APPROVED** [review]
Looks good!

File: src/api/metrics.rs [hunk 2]
```diff
@@ -28,27 +27,25 @@ impl MetricGetter {
...
-                metric_table.push(types::MetricTable {
-                    metric: metric.name.clone(),
-                    value: metric.value.clone(),
-                    step,
-                    timestamp,
-                });
-            }
+        for metric in metrics.metric.iter() {
+            let step = if metric.step.is_some() {
+                metric.step.as_ref().unwrap().to_string()
```
Comment (thetonus) [inline src/api/metrics.rs:32]: This is fine for this PR. But I suggest handling or just returning this error vs unwrapping. Given what `step` is (a deserialized value from serde), whatever error that is through will be a `serde` one.

Which means I think you should return a `Result<String, anyhow::Error>`.

Comment (thetonus) [reply src/api/metrics.rs:32]: To clarify, I think `.unwrap` for test code is okay. But for the application code, I think there should be a strategy to handle those errors vs panic.

**Review by thetonus — APPROVED** [review]

File: src/api/metrics.rs (hunk 1 above)
Comment (thorrester) [reply src/api/metrics.rs:95]: lol. same. I follow it as a convention

File: src/api/metrics.rs (hunk 2 above)
Comment (thorrester) [reply src/api/metrics.rs:32]: That's a good catch. I use that convention throughout, so I'll update this

**Pull request merged**